spanner/value.go

```diff
@@ -2921,14 +2926,21 @@ func decodeStruct(ty *sppb.StructType, pb *proto3.ListValue, ptr interface{}) er
 	if err != nil {
 		return ToSpannerError(err)
 	}
+	fieldNames := getAllFieldNames(v)
+	for _, f := range fieldNames {
+		sf := fields.Match(f)
+		if sf == nil {
+			return errDupGoField(ptr)
```

Comment:
Is this really a duplicate field? Or is it a missing field? I would expect `fields.Match(f)` to return nil if the field name is not in the list, and that this scenario is that the field is missing and not duplicate.

Reply:
Here it is duplicate field, it would be missing if we would have tried matching the spanner.Row field name, but here the same GO struct field names are matched from which the fields object was created.